### ISSUE TITLE
Evaluation of boolean-based blind SQLi on HTTP response headers

### ISSUE THREAD

**Issue by dcianciulli** [sqlmapproject/sqlmap#3518]
Is it possible to perform a boolean-based blind SQLi and distinct `True` from `False` based on Content-Length HTTP response header (or on any other response header)?
I tried with `--string`, `--not-string`, and `--regex` with no luck.

**Comment by stamparm** [comment]
It is automatically being included into the boolean-based blind comparison. `--string` or similar can only be used to distinguish TRUE from FALSE response, thus, if you know that certain `Content-Length` is for TRUE, you could use e.g. `--string="Content-length: 1024"`

You have to be more specific here so I could further advise you what to do. Currently I am just guessing what's your case about. Some excerpts from both T/F responses could help.

**Comment by dcianciulli** [comment]
Your guess matches perfectly my scenario.
I can't figure out why it does not work for me, probably I'm doing something wrong somewhere else.
Thank you very much.

**Comment by stamparm** [comment]
p.s. word of advise. If you are really into `Content-length` header please try both `--string="Content-length=..."` and `--string="Content-Length=..."`

**Comment by stamparm** [comment]
With the latest revision you'll be able to use exact capitalization as shown in higher verbosity levels (e.g. `-v 5`). Previously (e.g.) `Content-Length` has been forcefully capitalized to `Content-length` before showing in verbose output, while it stayed in original form when introduced to the boolean comparison engine.

TL;DR: run everything with `-v 5` and grep for `Content-Length` request header values. That same line (as is) can be used in `--string` (if your case is really a SQLi)
